discretisedfield/field.py

```diff
@@ -3475,3 +3476,117 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+        attribute of geometric dimensions is set to ``'m'``.
+
+        The name and units of the field ``DataArray`` can be set by using
+        ``name`` and ``units``. If the type of value passed to any of the two
+        arguments is not string, then a TypeError is raised.
+
+        Parameters
+        ----------
+        name : str, optional
+
+            String to set name of the field ``DataArray``
+
+        units : str, optional
+
+            String to set units of the field ``DataArray``
+
+        Returns
+        -------
+        xarray.DataArray
+
+            Field values DataArray.
+
+        Raises
+        ------
+        TypeError
+
+            If either ``name`` or ``units`` argument is not a string.
+
+        Examples
+        --------
+        >>> import discretisedfield as df
+        ...
+        >>> p1 = (0, 0, 0)
+        >>> p2 = (10, 10, 10)
+        >>> cell = (1, 1, 1)
+        >>> mesh = df.Mesh(p1=p1, p2=p2, cell=cell)
+        >>> field = df.Field(mesh=mesh, dim=3, value=(1, 0, 0), norm=1.)
+        ...
+        >>> field
+        Field(...)
+        ...
+        >>> field.to_xarray()
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10, comp: 3)>
+        array[...]
+        Coordinates:
+        * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * comp     (comp) <U1 'x' 'y' 'z'
+        ...
+        >>> field.to_xarray().sel(comp='x')
+        <xarray.DataArray 'field' (x: 10, y: 10, z: 10)>
+        Coordinates:
+        * x        (x) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * y        (y) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        * z        (z) float64 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5
+        comp     <U1 'x'
+        ...
+        >>> field.to_xarray(name=21)
+        Traceback (most recent call last):
+        ...
+        TypeError: name argument must be a string.
+        >>> field.to_xarray(units={'field': 'A/m'})
+        Traceback (most recent call last):
+        ...
+        TypeError: units argument must be a string.
+
+        """
+
+        if type(name) != str:
+            msg = "name argument must be a string."
+            raise TypeError(msg)
+        elif units is not None and type(units) != str:
+            msg = "units argument must be a string."
+            raise TypeError(msg)
+
+        geo_dim = ['x', 'y', 'z']
+
+        data_array_coords = {dim: np.fromiter(self.mesh.axis_points(dim),
+                                              dtype=float)
+                             for dim in geo_dim
+                             }
+
+        geo_units_dict = {dim: 'm' for dim in geo_dim}  # fix units for geo
+
+        if self.dim != 1:
+            data_array_dims = geo_dim + ['comp']
+            data_array_coords['comp'] = self.components
```

Comment:
`self.components` could be `None` (for `self.dim > 3` where users have to set the names explicitly [but are not forced to do so during initialisation because this could break some of the operators])

Reply:
Thank you mentioning this. I was not aware of it. I was under the impression that for `self.dim>3` it was commendatory to pass components list. This could have potentially broke `to_xarray` as one cannot set `None` as a dimension co-ordinate. I have changed the code accordingly.

Reply:
It is possible to create higher-dimensional fields implicitly and therefore `components` cannot be mandatory:
```python
>>> f1 = df.Field(..., dim=2, ...)  # has components x, y [automatically]
>>> f2 = df.Field(..., dim=2, ...)  # has components x, y [automatically]
>>> f1 << f2            # stack fields; we can neither pass component-labels manually
Field(..., dim=4, ...)  # nor use the existing ones as x, y are would appear twice.

```

Reply:
I understand, thank you for the example.